src/test/scala/org/broadinstitute/dsde/rawls/expressions/RhinoTest.scala

```diff
@@ -0,0 +1,707 @@
+
+      // Initialize the standard objects (Object, Function, etc.)
+      // This must be done before scripts can be executed.
+      val scope:Scriptable = cx.initStandardObjects();
+      val wrapped = Context.javaToJS(new Foo(), scope);
+      ScriptableObject.putProperty(scope, "foo", wrapped);
+
+      // Now we can evaluate a script. Let's create a new object
+      // using the object literal notation.
+//      val result = cx.evaluateString(scope, "function evaluate(thiz) { return thiz + \"bar!\" + foo.ret() }",
+      val result = cx.evaluateString(scope, "function evaluate(thiz) { return thiz + \"bar!\" + foo.ret() }",
+        "MySource", 1, null);
+
+      // TODO:
+      val eval = scope.get("evaluate", scope).asInstanceOf[Function];
+      val that = cx.newObject(scope);
+      println(eval.call(cx, scope, that, List("Foo").toArray))
+    } finally {
+      Context.exit();
+    }
+  }
+
+  test("rhino in the orient") {
+    val cx:Context = Context.enter();
+    try {
+      // Set version to JavaScript1.2 so that we get object-literal style
+      // printing instead of "[object Object]"
+      cx.setLanguageVersion(Context.VERSION_1_2);
+
+      // Initialize the standard objects (Object, Function, etc.)
+      // This must be done before scripts can be executed.
+      val scope:Scriptable = cx.initStandardObjects();
+
+      println("get the entity type from all samples from all sample sets from the root sampleSets")
+      val value = new QueryExecutor2(scope, graph).executeJS("function evaluate(value) {return <root.sampleSet.samples._entityType>}")("sampleSets")
+      println("results: " + value)
+      println()
+
+      val result = cx.evaluateString(scope, value,
+        "MySource", 1, null);
+
+      // TODO:
+      val eval = scope.get("evaluate", scope).asInstanceOf[Function];
+      val that = cx.newObject(scope);
+      println(eval.call(cx, scope, that, List("Foo").toArray))
+    } finally {
+      Context.exit();
+    }
+  }
+
+/* Disabled because a) doesn't work b) isn't in line with the ExpressionParsing class
+test("rhino in the orient II") {
+  val mgr:ScriptEngineManager  = new ScriptEngineManager();
+  // we are using the rhino javascript engine
+  val engine:ScriptEngine = mgr.getEngineByName("javascript");
+
+  try {
+    println("get the entity type from all samples from all sample sets from the root sampleSets")
+    val value = new QueryExecutor3(engine, graph).executeJS("function evaluate(value) {return <root.sampleSet.samples._entityType>}")("sampleSets")
+    println("results: " + value)
+    println()
+
+    val code = engine.eval(value)
+
+      val engineInv:Invocable = engine.asInstanceOf[Invocable];
+//        val func = engine.get("evaluate");
+      val result = engineInv.invokeFunction("evaluate", "foo");
+
+    println(result);
+  } finally {
+  }
+}
+*/
+
+class Foo() {
+  def ret() = { "var!" }
+}
+}
+
+class QueryExecutor2(scope:Scriptable, graph:Graph) extends JavaTokenParsers {
```

Comment:
Much of this is duplicated in ExpressionParsing now, but I decided to leave this here until we integrate Rhino properly.